src/pybind/mgr/cephadm/module.py

```diff
@@ -2375,6 +2426,137 @@ def upgrade_stop(self):
         self.event.set()
         return trivial_result('Stopped upgrade to %s' % target_name)
 
+    def save_spec(self, spec: ServiceSpec) -> None:
+        """
+        Attempts to save a ServiceSpec. Raises `OrchestratorError` if
+        an entry with the identifier already exist.
+
+        There are not two ways to manipulate the service_specs stored in the mon_store
+        1) Using a global definition (i.e. `ceph orch apply -i <specs_file.yaml>`)
+            This will usually contain multiple definitions of services and daemons.
+
+        2) Using the CLI for specific service types (i.e. `ceph orch apply rgw 3 --realm foo --zone bar`)
+
+        If these entries interfere, one would overwrite the other and vice versa.
+        Having the mentioned safeguard in place will prevent overwrites from two different sources.
+        """
+        json_spec = spec.to_json()
+        store = PersistentStoreDict(self, f'service_spec/{spec.service_type}')
+        name: str = spec.name or spec.service_type
+        try:
+            _ = store[name]
+            raise orchestrator.OrchestratorError(f"Specification for {name} already exists. "
+                                                 "Please review your existing specs with "
+                                                 "`ceph orch servicespecs ls` and try again.")
+        except IndexError:
+            self.log.info(f"New spec found. Saving <{name}> to the store.")
+            store[name] = json_spec
+
+    def list_specs(self) -> orchestrator.Completion:
+        """
+        Loads all entries from the service_spec mon_store root.
+        """
+        specs = list()
+        for spec_o in self.service_spec_store.items():
+            specs.append(spec_o)
+        return trivial_result(specs)
+
+    def clear_all_specs(self) -> orchestrator.Completion:
+        """
+        Clears the service_spec root in the mon_store (mgr/cephadm/service_specs/)
+        """
+        self.service_spec_store.clear()
+        return trivial_result(f"Mon store for {self.service_spec_store.prefix} cleared")
```

Comment:
this will be a potentially dangerous operations, of we decide on implementing daemon removal when deleting specs. I'd be :+1: for not adding this command right now.

Reply:
> this will be a potentially dangerous operations, of we decide on implementing daemon removal when deleting specs.

True, most convenient for development though :) I'll drop it in the next iteration